src/transactions/TransactionFrame.cpp

```diff
@@ -57,6 +59,83 @@ namespace
 // Limit to the maximum resource fee allowed for transaction,
 // roughly 112 million lumens.
 int64_t const MAX_RESOURCE_FEE = 1LL << 50;
+
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+// Starting in protocol 23, some operation meta needs to be modified
+// to be consumed by downstream systems. In particular, restoration is
+// logically a new entry creation from the perspective of ltx and stellar-core
+// as a whole, but this change type is reclassified to LEDGER_ENTRY_RESTORED
+// for easier consumption downstream.
+LedgerEntryChanges
+processOpLedgerEntryChanges(std::shared_ptr<OperationFrame const> op,
```

Comment:
While auto-restore is out of scope of this PR, I find it a bit concerning that this whole function is hardcoded to only work with restore op. Is there a generic way of converting the changes that is independent of the operation that has updated them (it's ok to have a top-level condition for the sake of performance though). I wonder if adding a restoration helper to ltx (as I've suggested above) could help with that.